### PR TITLE
Add support for diff-quality includes

### PR DESCRIPTION
**Pull request by kasium**

### CODE REVIEW

**Comment by kasium** [comment]
@Bachmann1234

Can you please support here. Locally the diff_cover/tests/test_snippets.py tests working fine (same command, same python version) but in the CI it fails.
But locally another test fails, which I don't understand even after debugging it for a while:

```plain
======================================================================================= FAILURES =======================================================================================
__________________________________________________________________ DiffQualityIntegrationTest.test_tool_not_installed __________________________________________________________________

self = <diff_cover.tests.test_integration.DiffQualityIntegrationTest testMethod=test_tool_not_installed>

    def test_tool_not_installed(self):
        # Pretend we support a tool named not_installed
        QUALITY_DRIVERS["not_installed"] = DoNothingDriver(
            "not_installed", ["txt"], ["not_installed"]
        )
        try:
            self._call_quality_expecting_error(
                "not_installed",
                "Failure: 'not_installed is not installed'",
>               report_arg="",
            )

diff_cover/tests/test_integration.py:585: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
diff_cover/tests/test_integration.py:568: in _call_quality_expecting_error
    logger.error.assert_called_with(expected_error)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

_mock_self = <MagicMock name='LOGGER.error' id='139827625166104'>, args = ("Failure: 'not_installed is not installed'",), kwargs = {}
expected = 'error("Failure: \'not_installed is not installed\'")'

    def assert_called_with(_mock_self, *args, **kwargs):
        """assert that the mock was called with the specified arguments.
    
        Raises an AssertionError if the args and keyword args passed in are
        different to the last call to the mock."""
        self = _mock_self
        if self.call_args is None:
            expected = self._format_mock_call_signature(args, kwargs)
>           raise AssertionError('Expected call: %s\nNot called' % (expected,))
E           AssertionError: Expected call: error("Failure: 'not_installed is not installed'")
E           Not called

/usr/lib64/python3.7/unittest/mock.py:811: AssertionError
```

**Comment by Bachmann1234** [comment]
I can take a closer look later this weekend!

File: diff_cover/diff_reporter.py [hunk 1]
```diff
@@ -72,20 +75,26 @@ def _is_path_excluded(self, path):
...
+            Path to check against the exclude and include patterns.
         :returns:
-            True if there are exclude patterns and the path matches,
-            otherwise False.
+            True if the patch should be excluded, otherwise False.
         """
+        absolute_path = os.path.abspath(path)
+        include = self._include
+        if include and not any(path in glob.glob(pattern, recursive=True) for pattern in include):
+            return True
```
Comment (Bachmann1234) [inline diff_cover/diff_reporter.py:89]: I like concise python statements as much as the next dev but this is too much for one line. The expended form feels like less load

```
if include:
  for pattern in include:
    if not any(path in glob.glob(pattern, recursive=True):
      return True
```

File: diff_cover/diff_quality_tool.py [hunk 2]
```diff
@@ -142,6 +143,10 @@ def parse_quality_args(argv):
         "--exclude", metavar="EXCLUDE", type=str, nargs="+", help=EXCLUDE_HELP
     )
 
+    parser.add_argument(
+        "--include", metavar="INCLUDE", default=["**/*.py"], nargs="+", type=str, help=INCLUDE_HELP
+    )
```
Comment (Bachmann1234) [inline diff_cover/diff_quality_tool.py:148]: @kasium This default is why the test is failing.

Nothing is matching the glob so the code never checks if the tool is installed. 

I also dont think this default is appropriate. This tools is not just used with python code. I think we should only be checking the glob when provided.

**Comment by Bachmann1234** [comment]
@kasium Just a heads up that I pushed a commit to master that fixes some tests due to an updated dependency. so if you update from upstream you should rebuild your virtual environment. 

A few of the HTML related tests are very fragile to the exact HTML output of a dep so sometimes I gotta update them when that dep updates.

File: diff_cover/diff_quality_tool.py (hunk 2 above)
Comment (kasium) [reply diff_cover/diff_quality_tool.py:148]: Default changed

**Comment by kasium** [comment]
Thanks a lot, let me give it a try

File: diff_cover/diff_reporter.py [hunk 3]
```diff
@@ -72,20 +75,28 @@ def _is_path_excluded(self, path):
...
-            otherwise False.
+            True if the patch should be excluded, otherwise False.
         """
+        absolute_path = os.path.abspath(path)
+        include = self._include
+        if include:
+            for pattern in include:
+                if not any(path in glob.glob(pattern, recursive=True)):
+                    return True
+
```
Comment (Bachmann1234) [inline diff_cover/diff_reporter.py:92]: So far none of the tests area actually hitting this code path. Given this is the core of the feature at least one would be good

**Comment by Bachmann1234** [comment]
@kasium This is looking pretty close. Thank you for your work on this! We can include the docs in this one or an upcoming one. The only thing holding me back from merging this and releasing it is the main functionality is still not tested. I left a comment over the code path that id like to see a test on.

**Comment by kasium** [comment]
@Bachmann1234 thanks for this missing point. I added now some tests for include/exclude combinations. I'm only used to pure pytest, so I hope the usage of tempfile is fine, because I don't want to mock glob completely.

File: diff_cover/diff_reporter.py (hunk 3 above)
Comment (kasium) [reply diff_cover/diff_reporter.py]: done

**Comment by Bachmann1234** [comment]
lgtm lets get this out

**Comment by Bachmann1234** [comment]
thank you for the pr!

**Comment by Bachmann1234** [comment]
Released! https://pypi.org/project/diff-cover/5.1.0/ Looking forward to the docs update

**Pull request merged**